Mathlib/Probability/Distributions/Gaussian.lean

```diff
@@ -347,6 +346,45 @@ lemma gaussianReal_mul_const {X : Ω → ℝ} (hX : Measure.map X ℙ = gaussian
 
 end Transformations
 
+open Measurable Real
+
+variable {Ω : Type*} {mΩ : MeasurableSpace Ω} {p : Measure Ω} {μ : ℝ} {v : ℝ≥0} {X : Ω → ℝ}
+
+lemma mgf_dirac (hX : p.map X = .dirac μ) (t : ℝ) : mgf X p t = exp (μ * t) := by
+  rw [← mgf_id_map, mgf, hX, integral_dirac, mul_comm]
+  · rfl
+  · apply AEMeasurable.of_map_ne_zero
+    rw [hX]
+    exact IsProbabilityMeasure.ne_zero (.dirac μ)
+
+theorem mgf_gaussianReal (hX : p.map X = gaussianReal μ v) (t : ℝ) :
+    mgf X p t = exp (μ * t + v * t ^ 2 / 2) := by
+  by_cases hv : v = 0
+  · simp only [gaussianReal, hv, ↓reduceIte] at hX
+    simp [mgf_dirac hX, hv]
+  calc
+  mgf X p t = (p.map X)[fun x => exp (t * x)] := by
+    rw [← mgf_id_map, mgf]
+    all_goals simp [AEMeasurable.of_map_ne_zero, hX, IsProbabilityMeasure.ne_zero]
+  _ = ∫ x, exp (t * x) * gaussianPDFReal μ v x := by
+    simp [hX, gaussianReal_of_var_ne_zero μ hv, gaussianPDF_def, ENNReal.ofReal,
+      integral_withDensity_eq_integral_smul (measurable_gaussianPDFReal μ v).real_toNNReal,
+      NNReal.smul_def, gaussianPDFReal_nonneg, mul_comm]
+  _ = exp (μ * t + v * t ^ 2 / 2)
+      * ∫ x, (√(2 * π * v))⁻¹ * exp (-(x - (μ + v * t)) ^ 2 / (2 * v)) := by
+    rw [gaussianPDFReal_def, ← integral_mul_left]
+    congr with x
+    field_simp only [mul_left_comm, ← exp_sub, ← exp_add]
+    ring_nf
```

Comment:
@YaelDillies , do you know how to do this in a single `simp` call? I tried but did not figure it out.